### PR TITLE
Support direct node reference

### PR DESCRIPTION
**Pull request by TharinduDG**
- Support direct node reference
- Support View Connection Property type
- making this a `SNAPSHOT` since more breaking changes are coming and we need to support changes ASAP
https://cognitedata.slack.com/archives/C031G8Y19HP/p1676390933465179
- some tests are ignored due to pending fdm bug fixes.

### CODE REVIEW

File: src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala [hunk 1]
```diff
@@ -130,7 +129,12 @@ class FlexibleDataModelsRelationTest extends FlatSpec with Matchers with SparkTe
                   .format(InstancePropertyValue.Timestamp.formatter)}' as timestampProp1,
                 |null as timestampProp2,
                 |'{"a": "a", "b": 1}' as jsonProp1,
-                |null as jsonProp2
+                |null as jsonProp2,
+                |named_struct(
+                |    'space', '$spaceExternalId',
+                |    'externalId', '$startNodeExtId'
+                |) as directRelation1,
```
Comment (silvavelosa) [inline src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala:136]: will this look like that in jetfire? or there will be a UDF for readability? @vuhainguyen

File: src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala [hunk 2]
```diff
@@ -362,20 +366,36 @@ class FlexibleDataModelsRelationTest extends FlatSpec with Matchers with SparkTe
...
-      viewExternalId = viewNodes.externalId,
-      viewVersion = viewNodes.version,
-      instanceSpaceExternalId = spaceExternalId
-    )
+//    val readNodesDf = readRows(
+//      viewSpaceExternalId = spaceExternalId,
+//      viewExternalId = viewNodes.externalId,
+//      viewVersion = viewNodes.version,
+//      instanceSpaceExternalId = spaceExternalId
+//    )
```
Comment (silvavelosa) [inline src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala:398]: should this be removed? or it's gonna be used later?

**Review by silvavelosa — APPROVED** [review]
Looks good.... just a question on usage, and a comment block

File: src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala (hunk 2 above)
Comment (silvavelosa) [reply src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala:398]: same for the blocks right bellow this one

Comment (TharinduDG) [reply src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala:398]: as mentioned in the description
```
 some tests are ignored due to pending fdm bug fixes.
```

File: src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala (hunk 1 above)
Comment (vuhainguyen) [reply src/test/scala/cognite/spark/v1/FlexibleDataModelsRelationTest.scala:136]: in jetfire we can use the UDF yes, Spark DS users have to use this redundant syntax instead

**Pull request merged**